src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs

```diff
@@ -287,4 +289,150 @@ public async Task SendAsync_WithZeroLengthHeaderName_Throws()
                 });
         }
     }
+
+    public abstract class HttpClientHandlerTest_Headers_NonAscii : HttpClientHandlerTestBase
+    {
+        public HttpClientHandlerTest_Headers_NonAscii() : base(null)
+        {
+        }
+
+        private static readonly (string Name, string[] Values)[] s_validLatin1Characters = new[]
+        {
+            ("Header1", new[] { "\x0081\x00FF\x00FE", "ascii-only" }),
+            ("Set-Cookie", new[] { "\u00FF", "\x00B0\x00B1\x00B2", "Ascii\x00B8" }),
+        };
+
+        private static readonly (string Name, string[] Values)[] s_invalidUnicodeCharacters = new[]
+        {
+            ("header-0", new[] { "\uD83D\uDE03", "\uD83D\uDE48\uD83D\uDE49\uD83D\uDE4A" }),
+            ("Set-Cookie", new[] { "\uD83C\uDDF8\uD83C\uDDEE" }),
+        };
+
+        private static (string Name, string[] Values)[] GetNonAsciiTestHeaders(bool unicode) => unicode ? s_invalidUnicodeCharacters : s_validLatin1Characters;
+
+        private static bool AllowLatin1Headers(string useAppCtxSwitchInner, string switchValueInner)
+        {
+            bool switchValue = bool.Parse(switchValueInner);
+            if (bool.Parse(useAppCtxSwitchInner))
+            {
+                AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.AllowLatin1Headers", switchValue);
+            }
+            else
+            {
+                Environment.SetEnvironmentVariable("DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_ALLOWLATIN1HEADERS", switchValueInner);
```

Comment:
Is this running in a child process? We generally don't want to set env vars in the test process as it affects other concurrent tests.

Reply:
Or AppContext, that's effectively static too.

Reply:
Right - for environment variables at least we would use RemoteInvoke. Otherwise I would expect potentially unreliable test results.

Reply:
Note: this got answered in a later https://github.com/dotnet/corefx/pull/42978#issuecomment-691257429